src/InputOutput/Writers/netcdf_writer.jl

```diff
@@ -18,8 +18,12 @@ function full_name(writer::NetCDFWriter, filename = nothing)
     end
 end
 
-function init_data(nc::NetCDFWriter, filename, dims, vars)
-    Dataset(full_name(nc, filename), "c") do ds
+function init_data(nc::NetCDFWriter, filename, no_overwrite, dims, vars)
+    nm = full_name(nc, filename)
+    if no_overwrite && ispath(nm)
+        error("$(nm) exists and overwriting is forbidden.")
```

Comment:
maybe there should be a warning thrown when overwriting an existing file as well

Reply:
just following the principal of "least surprise" when debugging  :)